packages/js-utils/src/unique-by.ts

```diff
@@ -0,0 +1,10 @@
+const defaultCompare = ( a: any, b: any ): boolean => a === b;
+
+function uniqueBy< T >( arr: T[], fn: ( a: T, b: T ) => boolean = defaultCompare ): T[] {
+	return arr.reduce( ( acc, v ) => {
+		if ( ! acc.some( ( x: any ) => fn( v, x ) ) ) acc.push( v );
```

Comment:
This looks like `O(n²)` in all cases. Via an `Object`/`Set` maybe we can make this to always be `O(n)`?

Reply:
A note that in this case we would need to move to a key function vs. the compare one, but given that we wanted to mimic `uniqBy` and not `uniqWith` that seems sensible trade-off.

Reply:
No worries!